### PR TITLE
Feature/improve errors

### LINKED ISSUE #80: Add failed step and scenario information to failure messages
When steps fail it can be difficult to understand exactly where in the test function things failed, especially on CI.

TTK already tries to help with this by adding a text attachment with lots of info when a failure occurs, but the log in the CI output can still be really unhelpful and just say something like "assert true failed".

I think we can improve this, since the most recently executed Scenario and Step are stored, including the line and file. We can add custom failure messages to XCTAssertTrue etc, and also potentially pass in the file and line from the most recent step (if there is one).

### PR DESCRIPTION
**Pull request by KaneCheshire**
Fixes #80

### CODE REVIEW

File: Example/Podfile.lock [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 SPEC CHECKSUMS:
-  ShowTime: 7821a8fc33629f34dbd970e0b55f34126f6d9750
-  TABTestKit: fd7beb2b5816d88d29c6ecb5af11ae47c3246e6e
+  ShowTime: 0d7ad97e415e3228f0e966e025087ea4175f3361
+  TABTestKit: 0e6828faebd9ee6fac09b992c9af7a78d20c8df4
 
 PODFILE CHECKSUM: cff5520fee18bc1ac482104a6673ae29b40d9690
 
-COCOAPODS: 1.9.0
+COCOAPODS: 1.8.4
```
Comment (KaneCheshire) [inline Example/Podfile.lock:27]: Curses, I'll try and re-run pod update on the correct cocoapods version (at least 1.9.0)

File: Example/TABTestKit_ExampleUITests/OtherElementsTests.swift [hunk 2]
```diff
@@ -131,7 +131,7 @@ final class OtherElementsTests: TABTestCase, SystemPreferencesContext {
     
     Scenario("Seeing and interacting with the switch") {
       Given(I: see(otherElementsScreen.toggle))
-      And(the: value(of: otherElementsScreen.toggle, is: .on))
+      And(the: value(of: otherElementsScreen.toggle, is: .off))
```
Comment (KaneCheshire) [inline Example/TABTestKit_ExampleUITests/OtherElementsTests.swift:134]: Temporary forced failure to see how Travis reports it

**Comment by KaneCheshire** [comment]
image
Much better CI reports!

**Comment by KaneCheshire** [comment]
And this is how that failure looks in Xcode
image

**Comment by KaneCheshire** [comment]
I think we could still do more to improve the actual messaging, there will still be places where it just say "XCTAssertTrue failed" but those failures will now show on the right line in the right file. I think that can be a separate PR though.

**Comment by zacoid55** [comment]
> I think we could still do more to improve the actual messaging, there will still be places where it just say "XCTAssertTrue failed" but those failures will now show on the right line in the right file. I think that can be a separate PR though.

Yeah I've seen it fail when the error message is longer than one line in some places so I think there's definitely something that can be expanded

**Comment by KaneCheshire** [comment]
Thanks! It's something that's been bothering me for a while

**Pull request merged**